src/dialogs/more-info/controls/more-info-weather.ts

```diff
@@ -0,0 +1,274 @@
+import "@polymer/iron-icon/iron-icon";
+import {
+  LitElement,
+  property,
+  CSSResult,
+  css,
+  customElement,
+} from "lit-element";
+import { HassEntity } from "home-assistant-js-websocket";
+import { TemplateResult, html } from "lit-html";
+
+import { HomeAssistant } from "../../../types";
+
+const cardinalDirections = [
+  "N",
+  "NNE",
+  "NE",
+  "ENE",
+  "E",
+  "ESE",
+  "SE",
+  "SSE",
+  "S",
+  "SSW",
+  "SW",
+  "WSW",
+  "W",
+  "WNW",
+  "NW",
+  "NNW",
+  "N",
+];
+
+const weatherIcons = {
+  "clear-night": "hass:weather-night",
+  cloudy: "hass:weather-cloudy",
+  exceptional: "hass:alert-circle-outline",
+  fog: "hass:weather-fog",
+  hail: "hass:weather-hail",
+  lightning: "hass:weather-lightning",
+  "lightning-rainy": "hass:weather-lightning-rainy",
+  partlycloudy: "hass:weather-partly-cloudy",
+  pouring: "hass:weather-pouring",
+  rainy: "hass:weather-rainy",
+  snowy: "hass:weather-snowy",
+  "snowy-rainy": "hass:weather-snowy-rainy",
+  sunny: "hass:weather-sunny",
+  windy: "hass:weather-windy",
+  "windy-variant": "hass:weather-windy-variant",
+};
+
+@customElement("more-info-weather")
+class MoreInfoWeather extends LitElement {
+  @property() public hass!: HomeAssistant;
```

Comment:
Should we update on every change of `hass` I think only on language change and `config.unit_system`?

Reply:
sure